tools/release-plugin.sh

```diff
@@ -158,21 +186,41 @@ if ! git push -u origin HEAD; then
 	die "Branch push failed. Check #jetpack-releases and make sure no one is doing a release already, then delete the branch at https://github.com/Automattic/jetpack/branches"
 fi
 
-yellow "Updating the changelogs."
-# Run the changelogger release script
-tools/changelogger-release.sh "${ARGS[@]}" "$PROJECT"
+# Loop through the projects and update the changelogs after building the arguments.
+for PLUGIN in "${!PROJECTS[@]}"; do
+	yellow "Updating the changelog files for $PLUGIN."
+	
+	if [[ ${ARGS} ]]; then
+		unset ARGS
+	fi
+
+	# Add the PR numbers to the changelog.
+	ARGS=('-p')
+	
+	# Add alpha and beta flags.
+	VERSION="${PROJECTS[$PLUGIN]}"
+	case $VERSION in
+		*-a* ) ARGS+=('-a');;
+		*-beta ) ARGS+=('-b');;
+	esac
+
+	# Explicitly pass the version number we want so there are no surprises.
+	ARGS+=( '-r' "${PROJECTS[$PLUGIN]}" )
+	ARGS+=("$PLUGIN");
+	tools/changelogger-release.sh "${ARGS[@]}"
+done
 
 # When it completes, wait for user to edit anything they want, then push key to continue.
 read -r -s -p $'Edit all the changelog entries you want (in a separate terminal or your text editor of choice (make sure to save)), then press enter when finished to continue the release process.'
 echo ""
 
 yellow "Committing changes."
 git add --all
-git commit -am "Changelog edits for $PROJECT"
+git commit -am "Changelog edits."
 
-# If we're running a beta, amend the changelog
-if [[ "$PROJECT" == "projects/jetpack" && "$ALPHABETA" == "-b" ]]; then
-	yellow "Releasing a beta, amending the readme.txt"
+# If we're releasing Jetpack and it's a beta, amend the readme.txt
+if [[ -v PROJECTS["plugins/jetpack"] && "${PROJECTS[plugins/jetpack]}" =~ '-beta' ]]; then
```

Comment:
Ooh, `-v` is even better than what I suggested back on line 83.

Probably better to test `== *-beta` instead of `=~ '-beta'` though, no need for regex when globbing will do.

Reply:
> Ooh, -v is even better than what I suggested back on line 83.

Ah yes, this is probably a chatGPT suggestion :D 

a7874b27a7425c07c5aa40c52cea3d47fb9adefb